lib/mixins/helpers.js

```diff
@@ -152,15 +139,14 @@ var helpers = {
       'slick-cloned': slickCloned
     });
   },
-  getListStyle: function () {
-    var style = {};
+  adaptHeight: function () {
     if (this.props.adaptiveHeight) {
       var selector = '[data-index="' + this.state.currentSlide +'"]';
       if (this.refs.list) {
-        style.height = this.refs.list.getDOMNode().querySelector(selector).offsetHeight;
+        var slickList = this.refs.list.getDOMNode();
+        slickList.style.height = slickList.querySelector(selector).offsetHeight + 'px';
```

Comment:
instead of setting `slickList.style.height = height`  like this,  extract the necessary css properties in componentDidMount and save them in state.  and use that state to calculate style

As you said, we can have markup like this in main render function.
and getStyle functions shouldl be purely based on state
```
<div>
   <List style={this.getListStyle()}>
       <Track style={this.getTrackStyle()}>
            {this.props.children}
       </Track>
   </List>
   <LeftArrow />
  <RightArrow />
 <Dots />
<div>
```